lib/api/funnel.js

```diff
@@ -167,7 +167,7 @@ class Funnel {
     // resolves the callback immediately if a slot is available
     if (this.concurrentRequests < this.kuzzle.config.limits.concurrentRequests) {
       if (this.requestsCacheById[request.internalId]) {
-        delete this.requestsCacheById[request.internalId];
+        this.requestsCacheById[request.internalId] = undefined;
```

Comment:
We need to keep the `delete` keyword here since the `requestsCacheById` will continue to grow with properties set to `undefined`.  
```
    const user = {
      name: 'aschen',
      age: 26
    };
    user.age = undefined;
    Object.keys(user) //=> ['name', 'age']
```

Reply:
or we can use a Map instead
either way that object will end up being treated as a b-tree, which is exactly what we want here